### PR TITLE
Rename `rcond/tol` to `rtol` in `linalg.matrix_rank` and `linalg.pinv`

### PR DESCRIPTION
**Pull request by Micky774**
Towards https://github.com/google/jax/issues/21088

This PR:
- Deprecates the `tol` argument of `linalg.matrix_rank` in favor of `rtol`
- Deprecates the `rcond` argument of `linalg.pinv` in favor of `rtol`

cc: @jakevdp

### CODE REVIEW

File: jax/_src/scipy/special.py [hunk 1]
```diff
@@ -599,7 +599,7 @@ def kl_div(
   .. math::
 
      \mathrm{kl\_div}(p, q) = \begin{cases}
-       p\log(p/q) & p>0,q>0\\
+       p\log(p/q)-p+q & p>0,q>0\\
```
Comment (jakevdp) [inline jax/_src/scipy/special.py:602]: Please rebase on an updated main branch to exclude unrelated changes

**Review by jakevdp — COMMENTED** [review]
Looks good!

File: jax/_src/scipy/special.py (hunk 1 above)
Comment (Micky774) [reply jax/_src/scipy/special.py]: Done!

**Review by jakevdp — APPROVED** [review]
Thanks!
